Update README.md and data-fetch example according to Hooks concept. (#8546)
zeit/next.js: packages/next/README.md

  </ul>
</details>

> When you need state, lifecycle hooks or **initial data population** you can export a [Class-Component](https://reactjs.org/docs/react-component.html) or use a Functional-Component [(a.k.a. Hooks)](https://reactjs.org/docs/hooks-intro.html).

Reviewer: ```suggestion
When you need state, lifecycle hooks or **initial data population** you can export a function component that uses [Hooks](https://reactjs.org/docs/hooks-intro.html) or [class component](https://reactjs.org/docs/react-component.html).
```

> Using a Functional-Component:

Reviewer: ```suggestion
Using a function component:
```

export default Page
```

> Using a Class-Component:

Reviewer: ```suggestion
Using a class component:
```

export default HelloUA
```

> Notice that to load data when the page loads, we use `getInitialProps` which is an [`async`](https://zeit.co/blog/async-and-await) method. It can asynchronously fetch anything that resolves to a JavaScript plain `Object`, which populates `props`.

Reviewer: ```suggestion
Notice that to load data when the page loads, we use `getInitialProps` which is an [`async`](https://zeit.co/blog/async-and-await) static method. It can asynchronously fetch anything that resolves to a JavaScript plain `Object`, which populates `props`.
```

It's still a static method, even when not set using the `static` keyword.
